blockly-core/core/ui/function_editor.js

```diff
@@ -721,6 +777,54 @@ Blockly.FunctionEditor.prototype.addCloseButton_ = function () {
   r.setAttribute('y', -bounds.height + padding - 1);
 };
 
+/**
+ * Add delete button to the top right of the modal dialog, to the
+ * left of the Close button.
+ * @private
+ */
+Blockly.FunctionEditor.prototype.addDeleteButton_ = function () {
+  if (!this.hasDeleteButton) {
+    return;
+  }
+  this.deleteButton_ = new Blockly.SvgTextButton(
+      this.modalBlockSpaceEditor.getSVGElement(), Blockly.Msg.DELETE,
+      this.onDeletePressed.bind(this));
+};
+
+Blockly.FunctionEditor.prototype.onDeletePressed = function () {
+  var functionName = this.functionDefinitionBlock.getProcedureInfo().name;
+  var deleteMessage = Blockly.Msg.CONFIRM_DELETE_FUNCTION_MESSAGE.replace('%1',
+      functionName);
+  if (!Blockly.customSimpleDialog) {
+    // For playground testing, use a standard `confirm` dialog for testing
```

Comment:
Would it make more sense to have the playground test override Blockly.showSimpleDialog rather than have test-only logic in our production code?